runtime/lua/vim/lsp/util.lua

```diff
@@ -1001,6 +1017,10 @@ function M.open_floating_preview(contents, filetype, opts)
 
   -- Compute size of float needed to show (wrapped) lines
   opts.wrap_at = opts.wrap_at or (vim.wo["wrap"] and api.nvim_win_get_width(0))
+  local close_events = opts.close_events or {"CursorMoved", "CursorMovedI", "BufHidden", "WinLeave"}
+  -- Unset opts.close_events before passing opts to other functions
+  opts.close_events = nil
```

Comment:
Why do we do this part, that seems a little weird.

Like if you passed the same opts to this function multiple times, it would do different things.

Reply:
alternatively, copying it and then removing would be OK for me.

Reply:
Because I have no formal CS education and no parent to smack me upside the head when I mess up :)

Yeah, I agree, it's weird, I'll change

Reply:
I removed the whole `close_events` logic now.